experiments/AMIP/modular/components/atmosphere/climaatmos_init.jl

```diff
@@ -50,33 +59,73 @@ get_field(sim::ClimaAtmosSimulation, ::Val{:turbulent_moisture_flux}) =
 
 get_field(sim::ClimaAtmosSimulation, ::Val{:thermo_state_int}) = Spaces.level(sim.integrator.p.ᶜts, 1)
 
+# extensions required by FluxCalculator (partitioned fluxes)
+get_field(sim::ClimaAtmosSimulation, ::Val{:height_int}) =
+    Spaces.level(Fields.coordinate_field(sim.integrator.u.c).z, 1)
+get_field(sim::ClimaAtmosSimulation, ::Val{:height_sfc}) =
+    Spaces.level(Fields.coordinate_field(sim.integrator.u.f).z, half)
+function get_field(sim::ClimaAtmosSimulation, ::Val{:uv_int})
+    uₕ_int = Geometry.UVVector.(Spaces.level(sim.integrator.u.c.uₕ, 1))
+    return @. StaticArrays.SVector(uₕ_int.components.data.:1, uₕ_int.components.data.:2)
+end
+get_field(sim::ClimaAtmosSimulation, ::Val{:air_density}) = TD.air_density.(thermo_params, sim.integrator.p.ᶜts)
+get_field(sim::ClimaAtmosSimulation, ::Val{:air_temperature}) = TD.air_temperature.(thermo_params, sim.integrator.p.ᶜts)
+get_field(sim::ClimaAtmosSimulation, ::Val{:cv_m}) = TD.cv_m.(thermo_params, sim.integrator.p.ᶜts)
+get_field(sim::ClimaAtmosSimulation, ::Val{:gas_constant_air}) =
+    TD.gas_constant_air.(thermo_params, sim.integrator.p.ᶜts)
+
+get_surface_params(sim::ClimaAtmosSimulation) = CAP.surface_fluxes_params(sim.integrator.p.params)
 
-function update_field!(sim::ClimaAtmosSimulation, ::Val{:surface_temperature}, field)
-    sim.integrator.p.radiation_model.surface_temperature .= RRTMGPI.field2array(field)
+# extensions required by the Interfacer
+function update_field!(sim::ClimaAtmosSimulation, ::Val{:surface_temperature}, csf)
+    sim.integrator.p.radiation_model.surface_temperature .= RRTMGPI.field2array(csf.T_S)
 end
+
 function update_field!(sim::ClimaAtmosSimulation, ::Val{:albedo}, field)
     sim.integrator.p.radiation_model.diffuse_sw_surface_albedo .=
         reshape(RRTMGPI.field2array(field), 1, length(parent(field)))
     sim.integrator.p.radiation_model.direct_sw_surface_albedo .=
         reshape(RRTMGPI.field2array(field), 1, length(parent(field)))
 end
 
-step!(sim::ClimaAtmosSimulation, t) = step!(sim.integrator, t - sim.integrator.t, true)
+# get_surface_params required by FluxCalculator (partitioned fluxes)
+function update_field!(sim::ClimaAtmosSimulation, ::Val{:turbulent_fluxes}, fields)
+    (; F_turb_energy, F_turb_moisture, F_turb_ρτxz, F_turb_ρτyz) = fields
 
-reinit!(sim::ClimaAtmosSimulation) = reinit!(sim.integrator)
+    Y = sim.integrator.u
+    surface_local_geometry = Fields.level(Fields.local_geometry_field(Y.f), Fields.half)
+    surface_normal = @. C3(unit_basis_vector_data(C3, surface_local_geometry))
 
-"""
-    update_sim!(atmos_sim::ClimaAtmosSimulation, csf)
+    vec_ct12_ct1 = @. CT12(CT2(unit_basis_vector_data(CT1, surface_local_geometry)), surface_local_geometry)
 
-Updates the surface fields for temperature and albedo.
+    vec_ct12_ct2 = @. CT12(CT2(unit_basis_vector_data(CT2, surface_local_geometry)), surface_local_geometry)
```

Comment:
Can you add a comment explaining what these two vectors are?

Reply:
sure!